src/Event/AbstractEvent.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy;
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
```

Comment:
Should the `composer.json` be updated to include a constraint for this new package?

Reply:
It will be available on 4.3 transitively. We should not require it ourselves I think because its only needed for Symfony 4.3+:

https://github.com/symfony/symfony/blob/4.3/src/Symfony/Component/EventDispatcher/composer.json#L20

Reply:
That's not great, but probably ok too…

Reply:
but this doesn't change the fact that class is coming from a library that sonata doesn't require and if it gets potentially removed as a dependency from event dispatcher, this code will break

Reply:
Well I can add another check `class_exists(ContractsEvent::class)`?

Reply:
Would be in line with what I see in other repos:

https://github.com/KnpLabs/knp-components/blob/master/src/Knp/Component/Pager/Event/Event.php

Reply:
I would say yes, but it's not my decision, cc @greg0ire